api/database/seeders/JobPosterTemplateSeeder.php

```diff
@@ -0,0 +1,187 @@
+        //       essentialBehaviouralSkillsNotes {
+        //         fr
+        //         en
+        //       }
+        //       essentialTechnicalSkillsNotes {
+        //         en
+        //         fr
+        //       }
+        //       nonessentialTechnicalSkillsNotes {
+        //         en
+        //         fr
+        //       }
+        //       workDescription {
+        //         en
+        //         fr
+        //       }
+        //       stream {
+        //         value
+        //       }
+        //       classification {
+        //         group
+        //         level
+        //       }
+        //       keywords {
+        //         en
+        //         fr
+        //       }
+        //     }
+        //   }
+
+        // Grab skills to create JSON
+        // query templatesSkills {
+        //     jobPosterTemplates {
+        //       referenceId
+        //       skills {
+        //         skill {
+        //           key
+        //         }
+        //         pivot {
+        //           type {
+        //             value
+        //           }
+        //           requiredLevel
+        //         }
+        //       }
+        //     }
+        //   }
+
+        $templatesFileContents = file_get_contents(base_path('database/seeders/JobPosterTemplateSeeder.data.json'));
+        if (! $templatesFileContents) {
+            throw new Exception('Failed to load Templates JSON file');
+        }
+        $templatesFileJson = json_decode($templatesFileContents);
+        if (! $templatesFileJson) {
+            throw new Exception('Failed to decode Templates JSON file');
+        }
+        $templatesModels = $templatesFileJson->data->jobPosterTemplates;
+        $skillsFileContents = file_get_contents(base_path('database/seeders/JobPosterTemplateSkillSeeder.data.json'));
+        if (! $skillsFileContents) {
+            throw new Exception('Failed to load Skills JSON file');
+        }
+        $skillsFileJson = json_decode($skillsFileContents);
+        if (! $skillsFileJson) {
+            throw new Exception('Failed to decode Skills JSON file');
+        }
+        $skillsModels = $skillsFileJson->data->jobPosterTemplates;
+
+        // Check for duplicate reference ids in templates
+        $referenceIds = array_map(
+            function ($model) {
+                return $model->referenceId;
+            },
+            $templatesModels
+        );
+        if (count(array_unique($referenceIds)) != count($templatesModels)) {
+            throw new Exception('The reference ids are not unique');
+        }
+
+        // TEMPLATES
+        // update or create all the templates
+        foreach ($templatesModels as $templateModel) {
+            $classificationObject = DB::table('classifications')
+                ->where('group', $templateModel->classification->group)
+                ->where('level', $templateModel->classification->level)
+                ->first();
```

Comment:
Could you fail here if one isn't found?  Like `sole` or `firstOrFail`?

Reply:
Sure https://github.com/GCTC-NTGC/gc-digital-talent/pull/11523/commits/2cebf8852851ff22c2a7f3347ffdbd0dce757e8b